airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -385,3 +398,25 @@ def state(self) -> MutableMapping[str, Any]:
     def state(self, value: StreamState):
         """State setter, accept state serialized by state getter."""
         self.stream_slicer.update_cursor(value)
+
+    def parse_records_and_emit_request_and_responses(self, request, response, stream_slice, stream_state) -> Iterable[AirbyteMessage]:
+        yield self._create_trace_message_from_request(request)
+        yield self._create_trace_message_from_response(response)
+        # Not great to need to call _read_pages which is a private method
+        # A better approach would be to extract the HTTP client from the HttpStream and call it directly from the HttpRequester
+        for record_mapping in self._read_pages(
+            lambda req, res, state, _slice: self.parse_response(res, stream_slice=_slice, stream_state=state), stream_slice, stream_state
+        ):
+            # FIXME: need to get the real json schema if we want to support type transforms
+            json_schema = {}
+            yield stream_data_to_airbyte_message(self.name, record_mapping, self.transformer, json_schema)
+
+    def _create_trace_message_from_request(self, request):
+        # FIXME: this should return some sort of trace message
+        log_message = f"request:{str(request)}"
```

Comment:
Will this print the complete request/response object? I was doing some debugging on the connector builder server work and when I emitted debug messages using `f"request:{str(request)}`, I would end up with `"request:<PreparedRequest [GET]>"` instead of the whole object